### PR TITLE
Add feed highlighting [Issue #1510]

### LINKED ISSUE #1510: Ability to highlight in feedlist using filter language
Related to #1104 

The behaviour of highlight-article does not seem to apply to the feedlist, so for example I cannot highlight a whole feed based on tags using filter language.

Example:
`highlight-article "tags # \"important\" and unread_count > 0" red default reverse bold`

With that all feeds deemed `important` would receive a major highlight when any new articles exist.

This has other obvious uses like being able to change the look of all feeds tagged `youtube` etc...

### PR DESCRIPTION
**Pull request by Vonter**
Hello!

I have tried to add feed highlighting as suggested in #1510 

If there's any suggestion of a better way this could be done, I would like to try it out.

Closes #1510

### CODE REVIEW

File: test/regexmanager.cpp [hunk 1]
```diff
@@ -227,8 +227,9 @@ TEST_CASE("quote_and_highlight wraps highlighted text in numbered tags",
 	}
 }
 
-TEST_CASE("RegexManager::dump_config turns each `highlight` and "
-	"`highlight-article` rule into a string, and appends them onto a vector",
+TEST_CASE("RegexManager::dump_config turns each `highlight`,"
+	"`highlight-article` and `highlight-feed` rule into a string, and appends"
+    "them onto a vector",
```
Comment (Minoru) [inline test/regexmanager.cpp:232]: Please put a space at the end of the string literal — when the literals are concatenated, no spaces are inserted, so we have to do it ourselves. In other words, `"hello"  "world"` becomes `"helloworld"`, so we do `"hello "  "world"` to get `"hello world"`.

**Review by Minoru — APPROVED** [review]
Excellent work! Please fix the formatting (see the failed CI job for the patch, and my comment for notes), and I'll be ready to merge this. The Coveralls job failed because Clang repo is being updated, I'll restart it tomorrow and it'll be good then.

Thanks so much for implementing this! Right in time too — the next release is just a couple weeks away :)

**Comment by Vonter** [comment]
Done. Thanks for reviewing the changes!

**Comment by Minoru** [comment]
Thanks!

Now the Coveralls job fails for a *yet another* reason: apparently Nightly changed the filename format, and our integration tests now create filenames that are too long (because Nightly concatenates the source filename with itself and some hash in between). I'll fix that in a different PR and then rebase and merge this one myself.

**Pull request merged**